cpp_tests/source/test_pspin.cpp

```diff
@@ -0,0 +1,48 @@
+#include "pele/array.h"
+#include "pele/combination.h"
+#include "pele/pspin_spherical.h"
+#include "pele/meta_pow.h"
+#include "test_utils.hpp"
+
+#include <iostream>
+#include <fstream>
+#include <stdexcept>
+#include <gtest/gtest.h>
+
+static double const EPS = std::numeric_limits<double>::min();
+#define EXPECT_NEAR_RELATIVE(A, B, T)  EXPECT_NEAR(A/(fabs(A)+fabs(B) + EPS), B/(fabs(A)+fabs(B) + EPS), T)
+
+using pele::Array;
+using pele::MeanFieldPSpinSpherical;
+using pele::pos_int_pow;
+
+class MeanFieldPSpinSphericalTest :  public PotentialTest
+{
+public:
+    size_t n;
+    Array<double> interactions;
+    virtual void setup_potential(){
+        pot = std::shared_ptr<pele::BasePotential> (new MeanFieldPSpinSpherical<4>(interactions, n));
+    }
+
+    virtual void SetUp(){
+        n = 20;
+        x = Array<double>(n+1,1);
```

Comment:
it's dangerous to have x be filled with ones.  It would miss any problem with the potential where you accidentally have an additional (or missing) factor of x[i]